api/admin/controller/sign_in.py

```diff
@@ -153,7 +153,7 @@ def _check_redirect(target: str) -> Tuple[bool, str]:
         we extract the URL path and forbid redirecting to external
         hosts.
         """
-        redirect_url: BaseURL = url_parse(target)
+        redirect_url = urlsplit(target)
```

Comment:
The deprecation message for `url_parse` recommended to replace it with a call to `urllib.parse.urlsplit`